crates/control/src/role_assignment.rs

```diff
@@ -155,198 +148,119 @@ impl RoleAssignment {
-            || cycle_start_time
-                .duration_since(self.last_transmitted_spl_striker_message.unwrap())?
-                > context.spl_network.spl_striker_message_send_interval;
+        // TODO: reimplement whatever this did
+        let is_in_penalty_kick = matches!(
+            context.filtered_game_controller_state,
+            Some(FilteredGameControllerState {
+                sub_state: Some(SubState::PenaltyKick),
+                ..
+            })
+        );
 
         let spl_striker_message_timeout = match self.last_received_spl_striker_message {
-            None => false,
+            None => true,
             Some(last_received_spl_striker_message) => {
                 cycle_start_time.duration_since(last_received_spl_striker_message)?
                     > context.spl_network.spl_striker_message_receive_timeout
             }
         };
 
-        let silence_interval_has_passed = match self.last_transmitted_spl_striker_message {
-            Some(last_transmitted_spl_striker_message) => {
-                cycle_start_time.duration_since(last_transmitted_spl_striker_message)?
-                    > context.spl_network.silence_interval_between_messages
-            }
-            None => true,
-        };
-
-        if send_game_controller_return_message {
-            self.last_system_time_transmitted_game_controller_return_message =
-                Some(cycle_start_time);
-            if let Some(address) = context.game_controller_address {
-                context
-                    .hardware
-                    .write_to_network(OutgoingMessage::GameController(
-                        *address,
-                        GameControllerReturnMessage {
-                            player_number: *context.player_number,
-                            fallen: matches!(context.fall_state, FallState::Fallen { .. }),
-                            pose: ground_to_field.as_pose(),
-                            ball: seen_ball_to_game_controller_ball_position(
-                                context.ball_position,
-                                cycle_start_time,
-                            ),
-                        },
-                    ))
-                    .wrap_err("failed to write GameControllerReturnMessage to hardware")?;
-            }
-        }
-
-        let mut team_ball = self.team_ball;
-
-        let is_in_penalty_kick = matches!(
-            context.filtered_game_controller_state,
-            Some(FilteredGameControllerState {
-                sub_state: Some(SubState::PenaltyKick),
-                ..
-            })
-        );
         if spl_striker_message_timeout && !is_in_penalty_kick {
             match new_role {
-                Role::Keeper => {
-                    team_ball = None;
-                }
-                Role::ReplacementKeeper => {
-                    team_ball = None;
-                }
+                Role::Keeper | Role::ReplacementKeeper => {}
```

Comment:
We should discuss the meaning/reason of this `match` statement and its implication. How can a striker message timeout and this very own robot be striker?!?! ... to then become loser... I'm confused